hcl2template/types.packer_config.go

```diff
@@ -401,6 +405,15 @@ func (cfg *PackerConfig) getCoreBuildPostProcessors(source SourceUseBlock, block
 			if moreDiags.HasErrors() {
 				continue
 			}
+
+			if cfg.bucket != nil && cfg.bucket.Validate() == nil {
+				postProcessor = &packer.RegistryPostProcessor{
```

Comment:
This post-processor wrapper gets created for all post-processors, not just the ones that create artifacts we want to track. For example, when I run the manifest postprocessor, the artifact that it returns is the same as the input artifact. this means that Packer tries to re-update/create the same build artifacts for the same region multiple times, resulting in an internal error `failed to update Packer registry with image artifacts` being returned because the build already exists and the attempted second update causes the database a duplicate key violation.

Reply:
repro case: 
```

data "amazon-ami" "whee-source" {
  filters = {
    name                = "ubuntu/images/*ubuntu-xenial-16.04-amd64-server-*"
    root-device-type    = "ebs"
    virtualization-type = "hvm"
  }
  most_recent = true
  owners      = ["099720109477"]
}

source "amazon-ebs" "packer-woo" {
  ami_name         = "packer-woo"
  force_deregister = true
  instance_type    = "t3.nano"
  source_ami       = "${data.amazon-ami.whee-source.id}"
  ssh_username     = "ubuntu"
  tags = {
    source = "{{ .SourceAMI }}"
  }
}

build {
  name = "packer-whee"
  sources = ["source.amazon-ebs.packer-woo"]

  provisioner "shell" {
    inline = ["echo hi"]
  }

  post-processor "manifest" {}
}

```

Remove the "manifest" pp and reset the fingerprint, and the build succeeds.

Reply:
This should be address in the latest changes.